src/NuGet.Clients/NuGet.PackageManagement.UI/ShowUpdatePackageOptions.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class ShowUpdatePackageOptions
+    {
+        public ShowUpdatePackageOptions(bool shouldUpdateAllPackages = false, IEnumerable<string> packagesToUpdate = null)
```

Comment:
It's possible to construct a nonsensical instance of this type (`true, new[]{"Foo"}`).

One option to prevent this is to make the constructor private and have static factory methods for the valid cases, with any validation unique to each therein. In this case, the "all" case could be a singleton.

Reply:
That's a great idea. Thanks and done!